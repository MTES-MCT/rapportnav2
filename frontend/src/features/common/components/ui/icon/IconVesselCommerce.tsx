import Icon from '../../../../../assets/images/navire_commerce.svg?url'

function IconVesselCommerce() {
  return <img src={Icon} alt="Navire de commerce" height={36} />
}

export default IconVesselCommerce
