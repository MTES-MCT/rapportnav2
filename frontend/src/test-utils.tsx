import React, { ReactElement } from 'react'
import { render, renderHook, RenderOptions } from '@testing-library/react'
import UIThemeWrapper from './ui/ui-theme-wrapper'
import { BrowserRouter } from 'react-router-dom'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <UIThemeWrapper>
      <BrowserRouter>{children}</BrowserRouter>
    </UIThemeWrapper>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

const customRenderHook = (hook: any, options?: Omit<RenderOptions, 'wrapper'>) =>
  renderHook(() => hook(), { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
export { customRenderHook as renderHook }
