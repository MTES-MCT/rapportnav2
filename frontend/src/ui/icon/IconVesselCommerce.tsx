import Icon from '../../assets/images/navire_commerce.svg'

function IconVesselCommerce() {
  return <img src={Icon} alt="Navire de commerce" />
}

export default IconVesselCommerce
