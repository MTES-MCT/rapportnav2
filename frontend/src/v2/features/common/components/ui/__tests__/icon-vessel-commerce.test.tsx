import { render } from '../../../../../../test-utils'
import IconVesselCommerce from '../icon-vessel-commerce'

describe('IconVesselCommerce', () => {
  it('should match the snapshot', () => {
    const wrapper = render(<IconVesselCommerce />)
    expect(wrapper).toMatchSnapshot()
  })
})
