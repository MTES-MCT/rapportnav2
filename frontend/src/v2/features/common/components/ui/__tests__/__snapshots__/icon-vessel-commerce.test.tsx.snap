// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconVesselCommerce > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="Navire de commerce"
        src="data:image/svg+xml,%3csvg%20height='36'%20viewBox='0%200%2069%2036'%20width='69'%20xmlns='http://www.w3.org/2000/svg'%3e%3cg%3e%3cpath%20d='m0%200h69v36h-69z'%20fill='transparent'/%3e%3cg%20fill='%233b4559'%3e%3cpath%20d='m61.985%2010.1h-61.985v-10.1h69l-5.333%207.738h-.042l-1.64%202.36zm-5.028-5.343a1.5%201.5%200%201%200%201.49%201.5%201.5%201.5%200%200%200%20-1.49-1.5zm-4.962%200a1.5%201.5%200%201%200%201.49%201.5%201.5%201.5%200%200%200%20-1.49-1.5zm-4.962%200a1.5%201.5%200%201%200%201.49%201.5%201.5%201.5%200%200%200%20-1.49-1.5z'%20transform='translate(.001%2025.9)'/%3e%3cpath%20d='m0%200h11v21h-11z'%20transform='translate(4.889%205.867)'/%3e%3cpath%20d='m17%205h-17v-5h17zm-5-4v3h3v-3zm-5%200v3h3v-3zm-5%200v3h3v-3z'%20transform='translate(1.889)'/%3e%3cpath%20d='m0%200h8v3h-8z'%20transform='translate(22.695%2023.756)'/%3e%3cpath%20d='m0%200h8v3h-8z'%20transform='translate(22.695%2019.797)'/%3e%3cpath%20d='m0%200h8v3h-8z'%20transform='translate(22.695%2015.836)'/%3e%3cpath%20d='m0%200h8v3h-8z'%20transform='translate(22.695%2011.877)'/%3e%3cpath%20d='m0%200h8v3h-8z'%20transform='translate(31.618%2023.756)'/%3e%3cpath%20d='m0%200h8v3h-8z'%20transform='translate(31.618%2019.797)'/%3e%3cpath%20d='m0%200h8v3h-8z'%20transform='translate(31.618%2015.836)'/%3e%3cpath%20d='m0%200h8v3h-8z'%20transform='translate(31.618%2011.877)'/%3e%3cpath%20d='m0%200h9v3h-9z'%20transform='translate(40.535%2023.756)'/%3e%3cpath%20d='m0%200h9v3h-9z'%20transform='translate(50.366%2023.756)'/%3e%3cpath%20d='m0%200h9v3h-9z'%20transform='translate(40.535%2019.797)'/%3e%3cpath%20d='m0%200h9v3h-9z'%20transform='translate(50.366%2019.797)'/%3e%3cpath%20d='m0%200h9v3h-9z'%20transform='translate(40.535%2015.836)'/%3e%3cpath%20d='m0%200h9v3h-9z'%20transform='translate(40.535%2011.877)'/%3e%3cpath%20d='m0%200h9v3h-9z'%20transform='translate(50.366%2015.836)'/%3e%3cpath%20d='m0%200h9v3h-9z'%20transform='translate(50.366%2011.877)'/%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
      />
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="Navire de commerce"
      src="data:image/svg+xml,%3csvg%20height='36'%20viewBox='0%200%2069%2036'%20width='69'%20xmlns='http://www.w3.org/2000/svg'%3e%3cg%3e%3cpath%20d='m0%200h69v36h-69z'%20fill='transparent'/%3e%3cg%20fill='%233b4559'%3e%3cpath%20d='m61.985%2010.1h-61.985v-10.1h69l-5.333%207.738h-.042l-1.64%202.36zm-5.028-5.343a1.5%201.5%200%201%200%201.49%201.5%201.5%201.5%200%200%200%20-1.49-1.5zm-4.962%200a1.5%201.5%200%201%200%201.49%201.5%201.5%201.5%200%200%200%20-1.49-1.5zm-4.962%200a1.5%201.5%200%201%200%201.49%201.5%201.5%201.5%200%200%200%20-1.49-1.5z'%20transform='translate(.001%2025.9)'/%3e%3cpath%20d='m0%200h11v21h-11z'%20transform='translate(4.889%205.867)'/%3e%3cpath%20d='m17%205h-17v-5h17zm-5-4v3h3v-3zm-5%200v3h3v-3zm-5%200v3h3v-3z'%20transform='translate(1.889)'/%3e%3cpath%20d='m0%200h8v3h-8z'%20transform='translate(22.695%2023.756)'/%3e%3cpath%20d='m0%200h8v3h-8z'%20transform='translate(22.695%2019.797)'/%3e%3cpath%20d='m0%200h8v3h-8z'%20transform='translate(22.695%2015.836)'/%3e%3cpath%20d='m0%200h8v3h-8z'%20transform='translate(22.695%2011.877)'/%3e%3cpath%20d='m0%200h8v3h-8z'%20transform='translate(31.618%2023.756)'/%3e%3cpath%20d='m0%200h8v3h-8z'%20transform='translate(31.618%2019.797)'/%3e%3cpath%20d='m0%200h8v3h-8z'%20transform='translate(31.618%2015.836)'/%3e%3cpath%20d='m0%200h8v3h-8z'%20transform='translate(31.618%2011.877)'/%3e%3cpath%20d='m0%200h9v3h-9z'%20transform='translate(40.535%2023.756)'/%3e%3cpath%20d='m0%200h9v3h-9z'%20transform='translate(50.366%2023.756)'/%3e%3cpath%20d='m0%200h9v3h-9z'%20transform='translate(40.535%2019.797)'/%3e%3cpath%20d='m0%200h9v3h-9z'%20transform='translate(50.366%2019.797)'/%3e%3cpath%20d='m0%200h9v3h-9z'%20transform='translate(40.535%2015.836)'/%3e%3cpath%20d='m0%200h9v3h-9z'%20transform='translate(40.535%2011.877)'/%3e%3cpath%20d='m0%200h9v3h-9z'%20transform='translate(50.366%2015.836)'/%3e%3cpath%20d='m0%200h9v3h-9z'%20transform='translate(50.366%2011.877)'/%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
